lib/plaid/connector.rb

```diff
@@ -7,30 +7,83 @@ module Plaid
   class Connector
     attr_reader :uri, :http, :request, :response, :body
 
+    # Private: Default read timeout for HTTP calls.
+    DEFAULT_TIMEOUT = 120
+
     # Private: Prepare to run request.
     #
-    # path - The String path without leading slash. E.g. 'connect'
-    def initialize(path, id = nil)
+    # path    - The String path without leading slash. E.g. 'connect'
+    # subpath - The String subpath. E.g. 'get'
+    # auth - The Boolean flag indicating that client_id and secret should be
+    #        included into the request payload.
+    def initialize(path, subpath = nil, auth: false)
       unless Plaid.env
         raise NotConfiguredError, "You must set Plaid.env before using any methods! Add a Plaid.config do .. end block somewhere in the initialization code of your program."
       end
 
+      if @auth = auth
+        cid, sec = Plaid.client_id, Plaid.secret
+
+        unless cid.is_a?(String) && cid.size > 0
+          raise NotConfiguredError, "You must set Plaid.client_id (currently it's #{cid.inspect}) before using any methods which require authentication!\nAdd a Plaid.config do .. end block somewhere in the initialization code of your program."
+        end
+
+        unless sec.is_a?(String) && sec.size > 0
+          raise NotConfiguredError, "You must set Plaid.secret (currently it's #{sec.inspect}) before using any methods which require authentication!\nAdd a Plaid.config do .. end block somewhere in the initialization code of your program."
+        end
+      end
+
       path = File.join(Plaid.env, path.to_s)
-      path = File.join(path, id) if id
+      path = File.join(path, subpath.to_s) if subpath
 
       @uri = URI.parse(path)
 
       @http = Net::HTTP.new(@uri.host, @uri.port)
       @http.use_ssl = true
+
+      @http.read_timeout = Plaid.read_timeout || DEFAULT_TIMEOUT
     end
 
     # Private: Run GET request.
```

Comment:
Many `Plaid::User`, `Plaid::Category`, and `Plaid::Institution` methods use `Connector` to make API calls, therefore `get`, `post`, `delete`, `patch` should be available as public.

But I made a digression from [TomDoc standard](http://tomdoc.org/), they should be marked as Internal, not Private. I'll rename everything.